### PR TITLE
Fixing percolation order

### PR DESCRIPTION
**Pull request by matthew7838**
This PR changes the order of operations of the soil water percolation and infiltration routines. Previously soil water would infiltrate the soil and then percolate through the layers. This causes some challenges with the soil water mass balance accounting.

Now percolation will happen at the begining of the day which will essentially percolate yesterday's soil water and then new water will be applied and infiltrated into the soil and through the soil layers.

### Context

Issue(s) closed by this pull request: closes #

- [x] The [changelog](https://docs.google.com/spreadsheets/d/1U-3igxdstHTFb6k5dVcE5-x9IL0r6q18eUiwwytKZyA/edit#gid=0) has been updated.
      
How big are the changes in the PR? Would you nominate it for a major version upgrade?
- [ ] Yes
- [x] No

### Why

Original order causes challenges with the soil water mass balance accounting.

### Test plan

Use this filter and assess the mass balance of the soil water additions and losses: 

[csv_soil_water.txt](https://github.com/RuminantFarmSystems/MASM/files/15490241/csv_soil_water.txt)

### CODE REVIEW

File: RUFAS/routines/field/soil/percolation.py
Comment (Isgotkowitz) [inline RUFAS/routines/field/soil/percolation.py]:
```suggestion
```
Since this routine is now running after the main percolation routine, there is nothing that needs to be returned by this method.

File: RUFAS/routines/field/field/field.py
Comment (Isgotkowitz) [inline RUFAS/routines/field/field/field.py:1394]: The test for `_cycle_water` needs to be updated.

File: RUFAS/routines/field/soil/soil.py
Comment (Isgotkowitz) [inline RUFAS/routines/field/soil/soil.py]: The second percolation routine should be called here (and the test will need to be updated).

**Review by Isgotkowitz — CHANGES_REQUESTED** [review]
I did a quick scan of the water content and percolation outputs, and I think they look good. Holding off on approving until the following things are addressed though: 
1. `percolate_infiltrated_water` should not return anything now.
2. All changes need to be tested.
3. The changelog needs to be updated.

Also, reading the PR description I was not able to tell how the changes made were evaluated (i.e. the Test Plan section was not filled in). Please include that when writing a PR description.

File: RUFAS/routines/field/soil/soil.py
Comment (matthew7838) [reply RUFAS/routines/field/soil/soil.py]: To clarify, you mean `percolate_infiltrated_water()`?

**Review by KFosterReed — APPROVED** [review]
I looked at the soil water balance between the water added in rainfall, irrigation, and snowmelt, the infiltrated water, and the percolated water in this updated branch and it aligns with what we were seeing before that the external SME team approved.

File: RUFAS/routines/field/soil/percolation.py
Comment (Isgotkowitz) [inline RUFAS/routines/field/soil/percolation.py:89]: Docstring should be changed as well.

**Review by Isgotkowitz — APPROVED** [review]
Thank you for addressing changes Matthew. Found one spot where a docstring needs to be updated, other than that LGTM

**Pull request merged**
